src/atoms/Button/Button.test.js

```diff
@@ -0,0 +1,73 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import Button from "./Button";
+
+describe("<Button />", () => {
+  const renderLikeSetup = (mockOnClick) => {
+    const onClick = jest.fn();
+    return render(
+      <Button
+        onClick={mockOnClick ?? onClick}
+        classFromProps="button--like"
+        buttonText="deixe seu like!"
+      />
+    );
+  };
+  const renderSubmitSetup = (mockOnClick) => {
+    const onClick = jest.fn();
+    return render(
+      <Button
+        onClick={mockOnClick ?? onClick}
+        buttonText="Adicionar"
+        classFromProps="button--submit"
+      />
+    );
+  };
+  it("renders a like button", () => {
+    renderLikeSetup();
+    const Button = screen.getByRole("button", { name: /like/i });
+    expect(Button).toBeInTheDocument();
+  });
+
+  it("renders a button with the correct submit class", () => {
+    renderSubmitSetup();
+    const submitButton = screen.getByRole("button", { name: /adicionar/i });
+    expect(submitButton).toHaveClass("button--submit");
+  });
+
+  it("renders a button with the correct like class", () => {
+    renderLikeSetup();
+    const Button = screen.getByRole("button", { name: /like/i });
+    expect(Button).toHaveClass("button--like");
+  });
+
+  it("calls the onClick function when clicked", () => {
+    const onClick = jest.fn();
+    renderLikeSetup(onClick);
+    const Button = screen.getAllByRole("button", { name: /like/i });
+    userEvent.hover(Button[0]);
+    userEvent.click(Button[0]);
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it("displays the correct text content", () => {
+    renderLikeSetup();
+    const Button = screen.getByRole("button", { name: /like/i });
+    expect(Button).toHaveTextContent(/DEIXE SEU LIKE!/i);
+  });
```

Comment:
2 pontos aqui:
1. Quando você passa o `name` para buscar o botão, você já meio que ta testando o "text content" mas ainda acho válido o texto, sempre bom garantir
2. Como está usando o `toHaveTextContent` daria para não passar o `name` para o botão

```suggestion
  it("displays the correct text content", () => {
    renderLikeSetup();
    const Button = screen.getByRole("button");
    expect(Button).toHaveTextContent(/DEIXE SEU LIKE!/i);
  });
```

Reply:
boooa!!! todas alteracoes feitas, fiquei naquela duvida acima la mesmo, vou te pingar disc